src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionposthearingapplication/ActionPostHearingApplicationAboutToSubmitHandler.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.actionposthearingapplication;
+
+import static java.util.Objects.requireNonNull;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.ActionPostHearingTypes;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DocumentGeneration;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DocumentStaging;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.resendtogaps.ListAssistHearingMessageHelper;
+import uk.gov.hmcts.reform.sscs.service.FooterService;
+import uk.gov.hmcts.reform.sscs.service.PostponementRequestService;
+import uk.gov.hmcts.reform.sscs.service.UserDetailsService;
+import uk.gov.hmcts.reform.sscs.util.SscsUtil;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ActionPostHearingApplicationAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
```

Comment:
Do we need this callback?

Reply:
Yes, for future tickets

Reply:
Future work should be done under the ticket when it comes time

Reply:
Hmm... I don't agree if it's laying the foundations for it. One of the reasons this is being done first so that the base structure is there and that the tickets coming after can just add to and existing class, rather than have 5 slightly different implementations of the same class.
Either way, its calling clearTransientFields so it is needed.